crates/orderbook/src/orderbook.rs

```diff
@@ -257,42 +252,37 @@ impl Orderbook {
         Metrics::on_order_operation(&old_order, OrderOperation::Cancelled);
         Metrics::on_order_operation(&new_order, OrderOperation::Created);
 
-        self.solvable_orders.request_update();
-
         Ok(new_order.metadata.uid)
     }
 
     pub async fn get_order(&self, uid: &OrderUid) -> Result<Option<Order>> {
-        let mut order = match self.database.single_order(uid).await? {
-            Some(order) => order,
-            None => return Ok(None),
-        };
-        set_available_balances(std::slice::from_mut(&mut order), &self.solvable_orders);
-        Ok(Some(order))
+        self.database.single_order(uid).await
     }
 
     pub async fn get_orders_for_tx(&self, hash: &H256) -> Result<Vec<Order>> {
-        let mut orders = self.database.orders_for_tx(hash).await?;
-        set_available_balances(orders.as_mut_slice(), &self.solvable_orders);
-        Ok(orders)
-    }
-
-    pub fn get_solvable_orders(&self) -> Result<SolvableOrders> {
-        let solvable_orders = self.solvable_orders.cached_solvable_orders();
-        ensure!(
-            solvable_orders.update_time.elapsed() <= self.solvable_orders_max_update_age,
-            "solvable orders are out of date"
-        );
-        Ok(solvable_orders)
+        self.database.orders_for_tx(hash).await
     }
 
-    pub fn get_auction(&self) -> Result<Auction> {
-        let (auction, update_time) = self.solvable_orders.cached_auction();
-        ensure!(
-            update_time.elapsed() <= self.solvable_orders_max_update_age,
-            "auction is out of date"
-        );
-        Ok(auction)
+    pub async fn get_auction(&self) -> Result<Option<Auction>> {
+        let auction = match self.database.most_recent_auction().await? {
+            Some(auction) => auction,
+            None => {
+                tracing::warn!("there is no current auction");
+                return Ok(None);
+            }
+        };
+        let current_block = self
+            .current_block
+            .borrow()
+            .number
+            .ok_or_else(|| anyhow!("no block number"))?
+            .as_u64();
+        let age_in_blocks = current_block.saturating_sub(auction.block);
+        if age_in_blocks > self.solvable_orders_max_update_age_blocks {
+            tracing::warn!("current auction is out of date");
+            return Ok(None);
+        }
```

Comment:
I wanted to have a way to represent "there is no current auction" which is why I added the Option to this function's return type and the 404 error in the corresponding api route handler.
After I had that it felt more natural to also do this when the auction is too old. The auction being too old isn't an internal server error from the point of view of the orderbook. I think the proper way to detect this error condition with the goal of having an alert is to set up a liveness check for the autopilot pod that takes this into account. I'll do that in a future PR.

Reply:
We could also completely get rid of the old age check here since I feel it has little purpose anymore. In that case we'd simply keep serving a potentially out of date auction while the autopilot fires an alert.

Reply:
I would be okay with dropping this check. Although I think I'm biased because I prefer to have fewer arguments to pass around (`solvable_orders_max_update_age_blocks`). Either way I don't feel strongly about that.

I agree that it makes sense to have a relaxed error handling in the orderbook if the autopilot is strict when alerting about errors while updating the auction.